test/libsolidity/semanticTests/events/event_emit_via_interface.sol

```diff
@@ -0,0 +1,13 @@
+interface I {
+    event Event(address indexed _from, uint256 _value);
+}
+
+contract C {
+    function emitEvent(uint256 _value) public {
+        emit I.Event(msg.sender, _value);
```

Comment:
Please also add a case that covers emitting an event defined in a foreign contract (i.e. one that you're not inheriting from). That's the other case enabled by #14274.

I'd also include an event of the same name in the emitting contract to make sure that such a name clash does not break anything. And do that in NatspecJSON tests too.

I think I'd also remove the CLI test. It does not seem like the right kind of test for something like this. It's currently testing a detail that we can reproduce just fine with more specific kinds of tests.

Finally, for completeness, we should also add ABI and ASTJSON tests. We skipped both of these in #14274.

Reply:
> I'd also include an event of the same name in the emitting contract to make sure that such a name clash does not break anything. And do that in NatspecJSON tests too.

Will do; I believe we already cover same name events defined in libraries, interfaces and contracts, but I'll have to double check.

> I think I'd also remove the CLI test. It does not seem like the right kind of test for something like this. It's currently testing a detail that we can reproduce just fine with more specific kinds of tests.

I mean, the assertion was thrown by requesting userdoc or devdoc; otherwise, it would not manifest, so I actually disagree with this to an extent; or are you saying that the boost test itself is enough in this case (they're do test the same code path more or less)?

> Finally, for completeness, we should also add ABI and ASTJSON tests.

Yikes, will do. I actually did create one, but then realized that the issue in here could not be reproduced with the ASTJSON test, so I just deleted it.

Reply:
> Will do; I believe we already cover same name events defined in libraries, interfaces and contracts, but I'll have to double check.

We do, but not in this specific case because it was not allowed before. I mean, it's not about testing such a clash in general. It's that for all these cases a name clash in Natspec is a distinct corner case we care about.

> I mean, the assertion was thrown by requesting userdoc or devdoc; otherwise, it would not manifest, so I actually disagree with this to an extent; or are you saying that the boost test itself is enough in this case (they're do test the same code path more or less)?

You mean it isn't. I haven't checked but yeah, those tests check the Natspec codepaths. I'd be very surprised if you really needed to run the whole compiler stack to make it show.

In any case, it's easy to check. Just put the tests in a separate commit, before the fix. If you run tests on that commit, you should see which ones fail and which one do not.

> Yikes, will do. I actually did create one, but then realized that the issue in here could not be reproduced with the ASTJSON test, so I just deleted it.

Those things are most likely fine. I'm only suggesting testing them as a way to get the test coverage we should have added in the original PR.

Reply:
> You mean it isn't. I haven't checked but yeah, those tests check the Natspec codepaths. I'd be very surprised if you really needed to run the whole compiler stack to make it show.

> In any case, it's easy to check. Just put the tests in a separate commit, before the fix. If you run tests on that commit, you should see which ones fail and which one do not.

Not sure I understand; what I'm saying is that I could reproduce the issue with both the boost (SolidityNatspecJSON) test and the cmdline test (running with vs without the assertion), so it's fine to remove the cmdline test if you say it's overkill. I'm just saying that it couldn't harm leaving it in. Is that what you meant?

Reply:
Ah, I forgot a question mark. "You mean it isn't?" was meant to be a question :) This was to confirm that I think the boost test is enough.

And yeah, no real harm having it there, just seemed superfluous. Command line tests are heavy and crude, so they just don't seem a particularly good way to test this. You can use them if you want, but we have other, perfectly good ways to test this and IMO those are sufficient here.

Reply:
Yup, that's what I was asking :) I deleted the cmdline test in any case, and added another natspec boost test, a couple of semantic tests, and an ASTJSON and an ABIJson test.